src/urllib3/connection.py

```diff
@@ -170,6 +171,9 @@ def _new_conn(self):
         if self.socket_options:
             extra_kw["socket_options"] = self.socket_options
 
+        if hasattr(sys, "audit"):
```

Comment:
This `hasattr` check gets called per connection, can we detect the feature once in the global scope and cache whether `sys.audit` is available? Let's also add a comment that mentions sys.audit being Python 3.8+

Reply:
Yeah, thanks for your feedback.
Will continue working on https://github.com/urllib3/urllib3/pull/2847. Will close this Pull Request.